libraries/Microsoft.Bot.Builder/ContextExtensions/ContextExtensionMethods.cs

```diff
@@ -2,20 +2,27 @@
 // Licensed under the MIT License.
 
 using Microsoft.Bot.Schema;
+using System.Collections.Generic;
+using System.Threading.Tasks;
 
 namespace Microsoft.Bot.Builder.ContextExtensions
 {
     public static class ContextExtensionMethods
     {
-        public static IBotContext ShowTyping(this IBotContext context)
+        public static async Task<IBotContext> ShowTyping(this IBotContext context)
         {
-            context.Responses.Add(new Activity { Type = ActivityTypes.Typing });
+            Activity activity = ((Activity)context.Request).CreateReply();
+            activity.Type = ActivityTypes.Typing;
+            await context.Bot.SendActivity(context, new List<IActivity>() { activity });
             return context;
         }
 
-        public static IBotContext Delay(this IBotContext context, int duration)
+        public static async Task<IBotContext> Delay(this IBotContext context, int duration)
         {
-            context.Responses.Add(new Activity { Type = "delay", Value = duration });
+            Activity activity = ((Activity)context.Request).CreateReply();
+            activity.Type = "delay";
+            activity.Value = duration;
+            await context.Bot.SendActivity(context, new List<IActivity>() { activity });
```

Comment:
I quite like that you've added these - I think they're very helpful. 

IF you can seperate them from the constant change in the Swagger layer, I think these are good to take. 

Now, to complicate things, I **think** we're going to change away from the queuing of responses, which are then run through a seperate adapter. It seems simpler all-the-way-around to simply have the Send methods just immediatly do a send. 

Doing that will make the framework work more how people expect, meaning we don't have to educate folks that "Send is not really send". That would be a big win. 

Until we make that change, however, (this week? next week?) these changes seem totally reasonable. 

Thanks~!!

Reply:
By separating from swagger do you mean by removing the ActivityTypes.Delay?

Reply:
Yes. The way you added the ActivityTypes.Delay just won't work given our auto-generation of the .Schema package and related code. 

This code, except for that, looks very reasonable at first glance.

Reply:
Okay cool. I have removed the Delay!